service/tcp.go

```diff
@@ -255,17 +241,58 @@ func (s *tcpService) handleConnection(listenerPort int, clientConn onet.DuplexCo
 				status = "ERR_REPLAY_CLIENT"
 			}
 			s.absorbProbe(listenerPort, clientConn, clientLocation, status, &proxyMetrics)
-			logger.Debugf(status+": %v in %s sent %d bytes", clientConn.RemoteAddr(), clientLocation, proxyMetrics.ClientProxy)
+			logger.Debugf(status+": %v in %s sent %d bytes", clientTCPConn.RemoteAddr(), clientLocation, proxyMetrics.ClientProxy)
 			return onet.NewConnectionError(status, "Replay detected", nil)
 		}
-		// Clear the authentication deadline
-		clientConn.SetReadDeadline(time.Time{})
 
 		ssr := ss.NewShadowsocksReader(clientReader, cipherEntry.Cipher)
+		tgtAddr, err := socks.ReadAddr(ssr)
+		if err != nil {
+			// Drain to prevent a close on cipher error.
+			io.Copy(ioutil.Discard, clientConn)
```

Comment:
This drain will run until the readtimeout, unlike errors later on, which will drain indefinitely.  Is that an intentional difference in behavior?